### ISSUE TITLE
[scrollable_positioned_list] How to change scrolling speed (question)

### ISSUE THREAD

**Issue by alexkmbk** [google/flutter.widgets#406]
Is it possible somehow to change the scrolling speed programmatically? (make it faster)

**Comment by nikitatg** [comment]
I've implemented it in my fork https://github.com/nikitatg/flutter.widgets by rev f10d88d235a1513897b1c12b3c4d32265fd01027

When building list can pass extra scroll speed like this

```dart
                  ScrollablePositionedList.builder(
                    itemPositionsListener: listener,
                    itemScrollController: _scrollController,
                    itemBuilder: (_, int i) => _messages[i],
                    itemCount: _messages.length,
                    reverse: widget.isReversed,
--->                extraScrollSpeed: SCROLL_EXTRA_SPEED,
                  ),
```
For me `const SCROLL_EXTRA_SPEED = 40;` (found empirically)

**Comment by alexkmbk** [comment]
> I've implemented it in my fork https://github.com/nikitatg/flutter.widgets by rev [f10d88d](https://github.com/google/flutter.widgets/commit/f10d88d235a1513897b1c12b3c4d32265fd01027)
> 
> When building list can pass extra scroll speed like this
> 
> ```dart
>                   ScrollablePositionedList.builder(
>                     itemPositionsListener: listener,
>                     itemScrollController: _scrollController,
>                     itemBuilder: (_, int i) => _messages[i],
>                     itemCount: _messages.length,
>                     reverse: widget.isReversed,
> --->                extraScrollSpeed: SCROLL_EXTRA_SPEED,
>                   ),
> ```
> 
> For me `const SCROLL_EXTRA_SPEED = 40;` (found empirically)

Yeah, it works, thanks!. I had to add some changes though, because I am still on flutter 2.10.4

**Comment by nikitatg** [comment]
@alexkmbk found an issue. Code in my fork not working correctly with touchscreen (when extraScrollSpeed != 0)

**Comment by alexkmbk** [comment]
> @alexkmbk found an issue. Code in my fork not working correctly with touchscreen (when extraScrollSpeed != 0). Makes kinetic scrolling impossible

thanks! are you considering to make a PR?

**Comment by nikitatg** [comment]
Definetly no, until I find a way to resolve touchscreens issue. And for now I have no ideas to do it on widget's level

**Comment by alexkmbk** [comment]
> Definetly no, until I find a way to resolve touchscreens issue. And for now I have no ideas to do it on widget's level

Let's then keep this issue open as soon as we don't have 100% working solution.

**Comment by nikitatg** [comment]
> @alexkmbk found an issue. Code in my fork not working correctly with touchscreen (when extraScrollSpeed != 0). Makes kinetic scrolling impossible

Hello. I have fixed it at rev [3d7071792f09aa7d66947266297440c2236105f9](https://github.com/nikitatg/flutter.widgets/commit/3d7071792f09aa7d66947266297440c2236105f9)

Now the widget just disables scroll acceleration when pointer device is recognized as touchscreen (`PointerDeviceKind.touch`)
Unfortunately there is no way to recognize if pointer device is notebook's touchpad (it is recognized as a mouse). For this reason accelerated scroll speed that was "ok" for mouse" most often will be too quick for touchpad scrolling.

<details>
  <summary>Scroll acceleration in other widgets</summary>
  
Any widget that are using ScrollController can be accelerated too. Below is the class for such purpose  
  ```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/gestures.dart';

const SCROLL_EXTRA_SPEED = 40;

class ScrollAcceleratorWrapper extends StatefulWidget {
  final Widget child;
  final ScrollController controller;

  const ScrollAcceleratorWrapper({
    Key? key,
    required this.child,
    required this.controller,
  }) : super(key: key);

  @override
  State<ScrollAcceleratorWrapper> createState() =>
      _ScrollAcceleratorWrapperState();
}

class _ScrollAcceleratorWrapperState extends State<ScrollAcceleratorWrapper> {
  bool _isTouchScreen = false;

  @override
  void initState() {
    speedUpScrolling(widget.controller, getIsTouchScreen: () => _isTouchScreen);
    super.initState();
  }

  // a way to speed up scroll taken from stackoverflow.com
  ScrollController speedUpScrolling(ScrollController controller,
      {bool Function()? getIsTouchScreen}) {
    controller.addListener(() {
      if (getIsTouchScreen != null && getIsTouchScreen()) {
        return;
      }
      ScrollDirection scrollDirection = controller.position.userScrollDirection;
      if (scrollDirection != ScrollDirection.idle) {
        double scrollEnd = controller.offset +
            (scrollDirection == ScrollDirection.reverse
                ? SCROLL_EXTRA_SPEED
                : -SCROLL_EXTRA_SPEED);
        scrollEnd = min(controller.position.maxScrollExtent,
            max(controller.position.minScrollExtent, scrollEnd));
        controller.jumpTo(scrollEnd);
      }
    });
    return controller;
  }

  @override
  Widget build(BuildContext context) => Listener(
        onPointerDown: (event) {
          // pointerDown triggers when touching screen (i guess before scroll began)
          _isTouchScreen = event.kind == PointerDeviceKind.touch;
        },
        onPointerHover: (event) {
          // pointerHover processing special to detect a mouse
          _isTouchScreen = event.kind == PointerDeviceKind.touch;
        },
        child: widget.child,
      );
}

  ```  
</details>

**Comment by alexkmbk** [comment]
> Any widget that are using ScrollController can be accelerated too

Yes, but the problem with the scrollable_positioned_list is that you cannot access the ScrollController from parent widget, so you have to change the class itself
